app/src/main/java/controlers/RestServices.java

```diff
@@ -0,0 +1,80 @@
+package controlers;
+
+import android.util.Base64;
+import android.widget.Toast;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import android.content.Context;
+
+import dto.Route;
+import dto.Stop;
+import dto.Departure;
+
+public class RestServices {
+    public JSONObject connectToRestService(String url_service, String param, Context context)
+    {
+        try {
+            if(url_service == null || url_service.length() == 0)
+                throw new Exception("URL invalid");
+
+            URL url = new URL(url_service);
+            HttpURLConnection connectionHttp = (HttpURLConnection) url.openConnection();
+            connectionHttp.setRequestMethod("POST");
+            connectionHttp.setRequestProperty("X-AppGlu-Environment", "staging");
+            connectionHttp.setRequestProperty("Content-Type", "application/json");
+            String base64EncodedCredentials = Base64.encodeToString(
+                    ("WKD4N7YMA1uiM8V:DtdTtzMLQlA0hk2C1Yi5pLyVIlAQ68").getBytes(), Base64.DEFAULT);
+            connectionHttp.setRequestProperty("Authorization", "Basic " + base64EncodedCredentials);
+
+            connectionHttp.setDoOutput(true);
+
+            if(param != null && param.length() > 0) {
+                byte[] outputInBytes = param.getBytes("UTF-8");
+                OutputStream os = connectionHttp.getOutputStream();
+                os.write(outputInBytes);
+                os.close();
+            }
+            connectionHttp.connect();
+            int responseCode = connectionHttp.getResponseCode();
+
+            if(responseCode >= 400)
+                throw new Exception("Unable to connect to rest service: " +  connectionHttp.getErrorStream());
+
+            InputStream inputStream = connectionHttp.getInputStream();
+
+            // Read the stream
+            byte[] b = new byte[1024];
+            ByteArrayOutputStream bAOS = new ByteArrayOutputStream();
+
+            while ( inputStream.read(b) != -1)
+                bAOS.write(b);
+
+            return new JSONObject(bAOS.toString());
```

Comment:
It's crashing sometimes here:

org.json.JSONException: Unterminated object at character 2028 of {"rows":[{"id":47,"name":"TICEN","sequence":1,"route_id":22},{"id":48,"name":"RUA PROCURADOR ABELARDO GOMES","sequence":2,"route_id":22},{"id":49,"name":"AVENIDA PAULO FONTES","sequence":3,"route_id":22},{"id":50,"name":"TERMINAL CIDADE DE FLORIANÓPOLIS","sequence":4,"route_id":22},{"id":51,"name":"RUA DOS ILHÉUS","sequence":5,"route_id":22},{"id":52,"name":"RUA VISCONDE DE OURO PRETO","sequence":6,"route_id":22},{"id":53,"name":"AVENIDA RIO BRANCO","sequence":7,"route_id":22},{"id":54,"name":"AVENIDA PREFEITO OSMAR CUNHA","sequence":8,"route_id":22},{"id":55,"name":"AVENIDA PROFESSOR OTHON GAMA D?EÇA","sequence":9,"route_id":22},{"id":56,"name":"AVENIDA JORNALISTA RUBENS DE ARRUDA RAMOS","sequence":10,"route_id":22},{"id":57,"name":"AVENIDA GOVERNADOR IRINEU BORNHAUSEN","sequence":11,"route_id":22},{"id":58,"name":"RUA COMANDANTE CONSTANTINO NICOLAU SPYRIDES","sequence":12,"route_id":22},{"id":59,"name":"RUA DELMINDA SILVEIRA","sequence":13,"route_id":22},{"id":60,"name":"AVENIDA PROFESSOR HENRIQUE DA SILVA FONTES","sequence":14,"route_id":22},{"id":61,"name":"TITRI","sequence":15,"route_id":22},{"id":62,"name":"RUA PROFESSOR MILTON ROQUE RAMOS KRIEGER","sequence":16,"route_id":22},{"id":63,"name":"RUA LAURO LINHARES","sequence":17,"route_id":22},{"id":64,"name":"RUA DELMINDA SILVEIRA","sequence":18,"route_id":22},{"id":65,"name":"RUA RUI BARBOSA","sequence":19,"route_id":22},{"id":66,"name":"RUA FREI CANECA","sequence":20,"route_id":22},{"id":67,"name":"RUA HEITOR LUZ","sequence":21,"route_id":22},{"id":68,"name":"RUA BOCAIÚVA","sequence":22,"route_id":22},{"id":69,"name":"AVENIDA PROFESSOR OTHON GAMA D?EÇA","sequence":23,"route_id":22},{"id":70,"name":"AVENIDA PREFEITO OSMAR CUNHA","sequence":24,"route_id":22},{"id":71,"name":"RUA JERÔNIMO COELHO","sequence":25,"route_id":22},{"id":72,"name":"RUA TENENTE SILVEIRA","sequence":26,"route_id":22},{"id":73,"name":"RUA PEDRO IVO","sequence":27,"route_id":22},{"id":74,"name":"AVENIDA PAULO FONTES","sequence":28,"route_id":22}],"rowsAffected":0OR HENRIQUE DA SIL}A FONTES","sequence":14,"route_id":22},{"id":61,"name":"TITRI","sequence":15,"route_id":22},{"id":62,"name":"RUA PROFESSOR MILTON ROQUE RAMOS KRIEGER","sequence":16,"route_id":22},{"id":63,"name":"RUA LAURO LINHARES","sequence":17,"route_id":22},{"id":64,"name":"RUA DELMINDA SILVEIRA","sequence":18,"route_id":22},{"id":65,"name":"RUA RUI BARBOSA","sequence":19,"route_id":22},{"id":66,"name":"RUA FREI CANECA","sequence":20,"route_id":22},{"id":67,"name":"RUA HEITOR LUZ","sequence":21,"route_id":22},{"id":68,"name":"RUA BOCAIÚVA","sequence":22,"route_id":22},{"id":69,"name":"AVENIDA PROFESSOR OTHON GAMA D?EÇA","sequence":23,"route_id":22},{"id":70,"name":"AVENIDA PREFEITO OSMAR CUNHA","sequence":24,"route_id":22},{"id":71,"name":"RUA JERÔNIMO COELHO","sequence":25,"route_id":22},{"id":72,"name":"RUA TENENTE SILVEIRA","sequence":26,"route_id":22},{"id":73,"name":"RUA PEDRO IVO","sequence":27,"route_id":22},{"id":74,"name":"AVENIDA PAULO FONTES","sequence":28,"route_id":22}],"rowsAffected":0OR HENRIQUE DA SIL